common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
+		this._setOptions(options);
+		if(this.client instanceof Realtime) {
+			this.client.connection.connectionManager.onAuthUpdated();
+		}
+	};
+
+	Auth.prototype.setClientId = function(clientId) {
+		if(clientId !== '*') {
+			if(this.clientId && this.clientId !== clientId) {
+				var msg = 'Unable to set incompatible clientId';
```

Comment:
> Can a user ever upgrade from no clientId i.e. user not yet logged in on their single page app to a connection with a clientId?

I agree we could consider that, but right now it is not supported. We shouldn't do it as part of this change IMO but address it when we do mutable auth where we can properly address the various upgrades that are possible.